kivy/config.py

```diff
@@ -911,7 +911,30 @@ def name(self, value):
             try:
                 _, section, name = key.split("_", 2)
             except ValueError:
-                raise
+                Logger.warning((
+                    "Environ `{}` invalid format, "
+                    "must be KCFG_section_name").format(key))
+                continue
+
+            # extract and check section
             section = section.lower()
+            if not Config.has_section(section):
+                Logger.warning(
```

Comment:
Also, since you're using Python logging under the hood, this is slightly an antipattern, as it evaluates the format even if warnings are suppressed.

Traditionally, logging uses :-/ old style % formatting, so what one is supposed to do is this:

    Logger.warning("Environ `%s`: unknown section `%s`", key, section)

If you hated % formatting enough not to want to do this, I'd totally understand.  :-D

Reply:
I understand (i didn't knew about it), but this is how we do logging in the whole Kivy library. We could however (not in this PR) make a pass to use the old style, dunno about it.